### ISSUE TITLE
Weekly Summary Reports

### ISSUE THREAD

**Issue by aritraroy** [louislam/uptime-kuma#1408]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

I have been trying out Uptime Kuma for a few days and have really liked it so far. However, I could not find any weekly summary report feature anywhere in the settings. By a weekly summary report, what I mean is a consolidated report of all the sites under monitoring and their performance over the week sent via email.

I have used other tools like betteruptime.com and have found this simple feature quite useful. Wanted to understand if this feature exists in UptimeKuma or is being planned?

### ✔️ Solution

A simple setting can be added which will allow users to opt-in for weekly summary reports. If they opt-in for it, then they will get a consolidated report of all their sites under monitoring and the performance for the week in an HTML-based email.

**Comment by aritraroy** [comment]
Any update on this feature request?

**Comment by CommanderStorm** [comment]
Blocked by implementing reports:
- `csv`: https://github.com/louislam/uptime-kuma/issues/1101
- `pdf`: https://github.com/louislam/uptime-kuma/issues/366

**Comment by CommanderStorm** [comment]
This can already be done via prometheus and writing a dashboard in grafana that displays exactly what you want.

https://github.com/louislam/uptime-kuma/wiki/Prometheus-Integration

**Comment by CommanderStorm** [comment]
After looking into this a bit, I noticed that this is actually a bit of a tricky requirement.
What users mean when they say "I want pdf uptime reports" is that they want to specify the accounting method (what is up/down, how are retries handled, ..), layout and display options.

Doing this well is not something where we have the resources for.

This can already be achived via using grafanas reporting tools.
This also enables users to fully customise what they want to display how.
https://github.com/louislam/uptime-kuma/wiki/Prometheus-Integration

=> closing as out of scope / wontfix / not planned for now
